bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/handler/IpCameraHandler.java

```diff
@@ -0,0 +1,1999 @@
+                                } else {
+                                    onvifCamera.sendPTZRequest("RelativeMoveUp");
+                                }
+                            } else {
+                                if ((boolean) config.get(CONFIG_PTZ_CONTINUOUS)) {
+                                    onvifCamera.sendPTZRequest("ContinuousMoveDown");
+                                } else {
+                                    onvifCamera.sendPTZRequest("RelativeMoveDown");
+                                }
+                            }
+                            return;
+                        } else if ("OFF".equals(command.toString())) {
+                            onvifCamera.sendPTZRequest("Stop");
+                            return;
+                        }
+                        onvifCamera.setAbsoluteTilt(Float.valueOf(command.toString()));
+                        scheduledMovePTZ.schedule(runnableMovePTZ, 500, TimeUnit.MILLISECONDS);
+                    }
+                    return;
+                case CHANNEL_ZOOM:
+                    if (onvifCamera.supportsPTZ()) {
+                        if (command instanceof IncreaseDecreaseType) {
+                            if ("INCREASE".equals(command.toString())) {
+                                if ((boolean) config.get(CONFIG_PTZ_CONTINUOUS)) {
+                                    onvifCamera.sendPTZRequest("ContinuousMoveIn");
+                                } else {
+                                    onvifCamera.sendPTZRequest("RelativeMoveIn");
+                                }
+                            } else {
+                                if ((boolean) config.get(CONFIG_PTZ_CONTINUOUS)) {
+                                    onvifCamera.sendPTZRequest("ContinuousMoveOut");
+                                } else {
+                                    onvifCamera.sendPTZRequest("RelativeMoveOut");
+                                }
+                            }
+                            return;
+                        } else if ("OFF".equals(command.toString())) {
+                            onvifCamera.sendPTZRequest("Stop");
+                            return;
+                        }
+                        onvifCamera.setAbsoluteZoom(Float.valueOf(command.toString()));
+                        scheduledMovePTZ.schedule(runnableMovePTZ, 500, TimeUnit.MILLISECONDS);
+                    }
+                    return;
+            }
+        }
+        // commands and refresh now get passed to brand handlers
+        switch (thing.getThingTypeUID().getId()) {
+            case "AMCREST":
+                AmcrestHandler amcrestHandler = new AmcrestHandler(getHandle());
+                amcrestHandler.handleCommand(channelUID, command);
+                if (lowPriorityRequests.isEmpty()) {
+                    lowPriorityRequests = amcrestHandler.getLowPriorityRequests();
+                }
+                break;
```

Comment:
All of these switch statements look like identical code. You could simplify this by having each camera handler implement a common interface:
```
public interface BrandHandler {
    public void handleCommand(ChannelUID channelUID, Command command);
    public List<String> getLowPriorityRequests();
}
```
```
    brandHandler.handleCommand(channelUID, command);
    if (lowPriorityRequests.isEmpty()) {
        lowPriorityRequests = brandHandler.getLowPriorityRequests();
    }
```

You might be able to refactor further by turning each of the brand handler classes into IPCameraHandler subclasses and just move their existing code into a nested classes. But you don't have to make such changes just yet if you don't want.

Reply:
I'm at a loss how to improve it as I am still learning Java. I have a huge amount to change still to meet the deadline of 2.5.9 build so I'll leave this till last.